src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -254,28 +321,31 @@ public static CallSite enumSwitch(MethodHandles.Lookup lookup,
         labels = labels.clone();
 
         Class<?> enumClass = invocationType.parameterType(0);
-        Stream.of(labels).forEach(l -> validateEnumLabel(enumClass, l));
-        MethodHandle temporary =
-                MethodHandles.insertArguments(DO_ENUM_SWITCH, 2, labels, lookup, enumClass, new ResolvedEnumLabels());
-        temporary = temporary.asType(invocationType);
+        labels = Stream.of(labels).map(l -> convertEnumConstants(lookup, enumClass, l)).toArray();
 
-        return new ConstantCallSite(temporary);
-    }
+        MethodHandle target;
+        boolean constantsOnly = Stream.of(labels).allMatch(l -> enumClass.isAssignableFrom(EnumDesc.class));
```

Comment:
This expression is always false, it should be:
```suggestion
        boolean constantsOnly = Stream.of(labels).allMatch(l -> l.isAssignableFrom(EnumDesc.class));
```